### ISSUE TITLE
Handle the different error sources from geometry::convhulln()

### ISSUE THREAD

**Issue by Rekyt** [Bisaloo/fundiversity#16]
95e582350b6766b9b37253f6c2338c68513e3a48 fix the case when geometry::convhulln() returns an error if there are less points than dimensions.
We should also handle the case where the dimensionality is artificially reduced because several points have the same coordinates.

**Comment by Rekyt** [comment]
This it what happens when we have duplicates points that do not increase the actual dimensionality of the dataset:
``` r
library("fundiversity")
data(traits_birds)
df = dplyr::bind_rows(lapply(1:10, function(x) {
  as.data.frame(traits_birds[1,,drop=FALSE])
  }))

fd_fric(df)
#> Error in geometry::convhulln(traits, "FA"): Received error code 1 from qhull. Qhull error:
#> QH6013 qhull input error: input is less than 4-dimensional since it has the same x coordinate
#> 
#> While executing:  | qhull FA  Qt
#> Options selected for Qhull 2015.2.r 2016/01/18:
#>   run-id 1408970596  FArea-total  Qtriangulate  _pre-merge  _zero-centrum
#>   _max-width  0  Error-roundoff 1.6e-12  _one-merge 1.5e-11
#>   _near-inside 7.3e-11  Visible-distance 9.7e-12  U-coplanar-distance 9.7e-12
#>   Width-outside 1.9e-11  _wide-facet 5.8e-11
```

<sup>Created on 2021-01-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

The easiest fix I see is to remove duplicate trait values before using fd_fric_single()

**Comment by Rekyt** [comment]
> [95e5823](https://github.com/Bisaloo/fundiversity/commit/95e582350b6766b9b37253f6c2338c68513e3a48) fix the case when geometry::convhulln() returns an error if there are less points than dimensions.
> We should also handle the case where the dimensionality is artificially reduced because several points have the same coordinates.

Fixed by 1712a07
